src/lib/comms/sol-socket-impl-zephyr.c

```diff
@@ -0,0 +1,423 @@
+/*
+ * This file is part of the Soletta Project
+ *
+ * Copyright (C) 2015 Intel Corporation. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include <errno.h>
+
+#include <net/ip_buf.h>
+#include <sections.h>
+
+#include "sol-buffer.h"
+#include "sol-log.h"
+#include "sol-mainloop.h"
+#include "sol-mainloop-zephyr.h"
+#include "sol-socket-impl.h"
+#include "sol-vector.h"
+#include "sol-network-util.h"
+
+struct sol_socket_net_context;
+
+struct sol_socket_zephyr {
+    struct sol_socket base;
+
+    struct {
+        bool (*cb)(void *data, struct sol_socket *s);
+        const void *data;
+    } read, write;
+
+    struct sol_timeout *write_timeout;
+
+    struct sol_socket_net_context *unicast_context;
+    struct sol_ptr_vector mcast_contexts;
```

Comment:
So, in the end we decided to leave it as is to be coherent with how Linux and RIOT work regarding order of bind() and join_group(), right?